src/Evolution/Systems/GeneralizedHarmonic/DampedHarmonicGaugeQuantities.cpp

```diff
@@ -0,0 +1,928 @@
+struct D4LogFacBuffer;
+
+template <size_t SpatialDim, typename Frame>
+struct D4LogFacBuffer<SpatialDim, Frame, double> {
+  // NOLINTNEXTLINE(cppcoreguidelines-pro-type-member-init)
+  explicit D4LogFacBuffer(const size_t /*size*/) noexcept {}
+
+  tnsr::a<double, SpatialDim, Frame> d_g;
+  Scalar<double> d0_N;
+  tnsr::i<double, SpatialDim, Frame> d3_N;
+  tnsr::a<double, SpatialDim, Frame> d_N;
+};
+
+template <size_t SpatialDim, typename Frame>
+struct D4LogFacBuffer<SpatialDim, Frame, DataVector> {
+  private:
+    // We make one giant allocation so that we don't thrash the heap.
+    Variables<tmpl::list<::Tags::Tempa<0, SpatialDim, Frame>,
+                        ::Tags::TempScalar<1>,
+                        ::Tags::Tempi<2, SpatialDim, Frame>,
+                        ::Tags::Tempa<3, SpatialDim, Frame>>> buffer_;
+
+  public:
+    explicit D4LogFacBuffer(const size_t size) noexcept : buffer_(size),
+      d_g(get<::Tags::Tempa<0, SpatialDim, Frame>>(buffer_)),
+      d0_N(get<::Tags::TempScalar<1>>(buffer_)),
+      d3_N(get<::Tags::Tempi<2, SpatialDim, Frame>>(buffer_)),
+      d_N(get<::Tags::Tempa<3, SpatialDim, Frame>>(buffer_)) {}
+
+    tnsr::a<DataVector, SpatialDim, Frame>& d_g;
+    Scalar<DataVector>& d0_N;
+    tnsr::i<DataVector, SpatialDim, Frame>& d3_N;
+    tnsr::a<DataVector, SpatialDim, Frame>& d_N;
+};
+}  // namespace
+
+template <size_t SpatialDim, typename Frame, typename DataType>
+tnsr::a<DataType, SpatialDim, Frame> spacetime_deriv_of_log_factor(
+    const Scalar<DataType>& lapse,
+    const tnsr::I<DataType, SpatialDim, Frame>& shift,
+    const tnsr::A<DataType, SpatialDim, Frame>& spacetime_unit_normal,
+    const tnsr::II<DataType, SpatialDim, Frame>& inverse_spatial_metric,
+    const Scalar<DataType>& sqrt_det_spatial_metric,
+    const tnsr::ii<DataType, SpatialDim, Frame>& dt_spatial_metric,
+    const tnsr::aa<DataType, SpatialDim, Frame>& pi,
+    const tnsr::iaa<DataType, SpatialDim, Frame>& phi,
+    const double& exponent) noexcept {
+  // Use a Variables to reduce total number of allocations. This is especially
+  // important in a multithreaded environment.
+  D4LogFacBuffer<SpatialDim, Frame, DataType> buffer(get_size(get(lapse)));
+  // Get \f$ \partial_a g\f$
+  buffer.d_g = spacetime_deriv_of_det_spatial_metric<SpatialDim, Frame,
+    DataType>(sqrt_det_spatial_metric, inverse_spatial_metric,
+      dt_spatial_metric, phi);
+  // Get \f$ \partial_a N\f$
+  buffer.d0_N = time_deriv_of_lapse<SpatialDim, Frame, DataType>(lapse,
+    shift, spacetime_unit_normal, phi, pi);
+  buffer.d3_N = spatial_deriv_of_lapse<SpatialDim, Frame, DataType>(lapse,
+    spacetime_unit_normal, phi);
+  get<0>(buffer.d_N) = get(buffer.d0_N);
+  for (size_t i = 0; i < SpatialDim; ++i) {
+    buffer.d_N.get(1 + i) = buffer.d3_N.get(i);
+  }
+  // Compute
+  auto d_logfac = make_with_value<tnsr::a<DataType, SpatialDim, Frame>>(
+    get(lapse), 0.);
+  const auto one_over_lapse = 1. / get(lapse);
```

Comment:
In a `compute_spacetime_deriv_damped_harmonic_h` you also have `one_over_lapse` in the buffer. Why not here?

Reply:
it should've been, added it. thanks!